trace-api-plugin/trace_api_plugin_enhancements

```diff
@@ -0,0 +1,20 @@
+# Trace API Plugin enhancements
+
+## ABI Support
+
+Currently `trace_api_plugin` only supports decoding `action` and `return_value` data via command line provided `trace-rpc-abi` files. This requires manually 
+providing new ABI file any time the ABI changes.
+Add using on-chain ABI at time of action execution for decoding of the `action` and `return_value` data.
+Provide option to not track ABI if user does not require data expanded via ABI. ABI decoding should be the default.
+
+## Index transactions by transaction id for quicker retrieval.
+
+Currently `trace_api_plugin` does a sequential scan for `/v1/trace_api/get_transaction_trace`. Add an index for faster lookup.
+
+## Investigate removing `fsync`
+
+We do not use `fsync` for other file writing. Determine if `fflush` is sufficient.
+
+## Use SQLite for storage
+
+Usage would simplify the index on transaction issue and provide easier storage/lookup of ABI. Also solves the `fsync` question.
```

Comment:
Does it solve it or just move it? I thought (by default) a transaction in sqlite implies an fsync. So if, for example, we do an sqlite transaction for each block, it'd be no different in fsync overhead as before.

Reply:
https://www.sqlite.org/pragma.html#pragma_synchronous
By default `fsync` is still used. So move is a better description. It does look like it can be smarter about the `fsync` than our current implementation. Although, our current implementation likely doesn't even need it.

Reply:
From our 11/8/2022 conversation on this topic:

- need to understand potential impacts on transaction processing time
- might explore alternative proposal adding filtering capabilities to SHiP.